lib/ansible/module_utils/aws/rds.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Author: Michael De La Rue 2017 largely rewritten but based on work
+# by Will Thames taken in turn from the original rds module.
+
+try:
+    import botocore
+    from botocore import xform_name
+except:
+    pass  # it is assumed that calling modules will detect and provide an appropriate nice error.
+
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
+
+DEFAULT_PORTS = {
+    'aurora': 3306,
+    'mariadb': 3306,
+    'mysql': 3306,
+    'oracle': 1521,
+    'sqlserver': 1433,
+    'postgres': 5432,
+}
+
+DB_ENGINES = [
+    'aurora',
+    'mariadb',
+    'mysql',
+    'oracle-ee',
+    'oracle-se',
+    'oracle-se1',
+    'oracle-se2',
+    'postgres',
+    'sqlserver-ee',
+    'sqlserver-ex',
+    'sqlserver-se',
+    'sqlserver-web',
+]
+
+LICENSE_MODELS = [
+    'bring-your-own-license',
+    'general-public-license',
+    'license-included',
+    'postgresql-license'
+]
+
+
+def get_db_instance(conn, instancename):
+    """return AWS format DB instance
+
+    This function connects to AWS and retrieves the information about
+    the instance in directly in the standard AWS format.
+    """
+    # FIXME: currently this makes two API calls, one for the instance and
+    # one for the tags.  This mostly make sense since the object we get
+    # back is close to what we want to call a create with, but for
+    # efficiency we should maybe put an option to not make the tags call in
+    # the case where we throw that data away.
+    try:
+        response = conn.describe_db_instances(DBInstanceIdentifier=instancename)
```

Comment:
This doesn't get used anywhere yet. I feel like it should either be used (you could add DBInstanceIdentifier optionally to a params dict and paginate instead of re-coding the describe_db_instances and tags transformation in rds_instance_facts) or else maybe it should be removed until it gets used. I also feel like get_snapshot should be removed for now since it doesn't handle aurora and nothing is using it. But I care less about these two straightforward functions than instance_facts_diff(), which I definitely think should be removed until it is used and tested.

Reply:
Yeah, I think this gets used by rds_instance (spoiler!) but can delete along with instance_facts_diff until then.